src/e2e-test/java/io/cdap/plugin/gcs/stepsdesign/GCSConnector.java

```diff
@@ -0,0 +1,428 @@
+    for (WebElement element : previewOutputSchemaElements) {
+      previewOutputSchema.add(element.getAttribute("title"));
+    }
+    Assert.assertTrue(previewOutputSchema.equals(propertiesOutputSchema));
+  }
+
+  @Then("Close the Preview")
+  public void closeThePreview() {
+    CdfGCSLocators.closeButton.click();
+    CdfStudioActions.previewSelect();
+  }
+
+  @Then("Deploy the pipeline")
+  public void deployThePipeline() {
+    SeleniumHelper.waitElementIsVisible(CdfStudioLocators.pipelineDeploy, 2);
+    CdfStudioActions.pipelineDeploy();
+  }
+
+  @Then("Save and Deploy Pipeline")
+  public void saveAndDeployPipeline() throws InterruptedException {
+    CdfStudioActions.pipelineName();
+    CdfStudioActions.pipelineNameIp("TestPipeline" + UUID.randomUUID().toString());
+    CdfStudioActions.pipelineSave();
+    SeleniumHelper.waitElementIsVisible(CdfStudioLocators.statusBanner);
+    WebDriverWait wait = new WebDriverWait(SeleniumDriver.getDriver(), 5);
+    wait.until(ExpectedConditions.invisibilityOf(CdfStudioLocators.statusBanner));
+    CdfStudioActions.pipelineDeploy();
+  }
+
+  @Then("Run the Pipeline in Runtime")
+  public void runThePipelineInRuntime() throws InterruptedException {
+    CdfPipelineRunAction.runClick();
+  }
+
+  @Then("Wait till pipeline is in running state")
+  public void waitTillPipelineIsInRunningState() throws InterruptedException {
+    WebDriverWait wait = new WebDriverWait(SeleniumDriver.getDriver(), 300);
+    wait.until(ExpectedConditions.or(
+      ExpectedConditions.visibilityOfElementLocated(By.xpath("//*[@data-cy='Succeeded']")),
+      ExpectedConditions.visibilityOfElementLocated(By.xpath("//*[@data-cy='Failed']"))));
+  }
+
+  @Then("Verify the pipeline status is {string}")
+  public void verifyThePipelineStatusIs(String status) {
+    boolean webelement = false;
+    webelement = SeleniumHelper.verifyElementPresent("//*[@data-cy='" + status + "']");
+    Assert.assertTrue(webelement);
+  }
+
+  @Then("Open Logs")
+  public void openLogs() throws FileNotFoundException, InterruptedException {
+    CdfPipelineRunAction.logsClick();
+    BeforeActions.scenario.write(CdfPipelineRunAction.captureRawLogs());
+    PrintWriter out = new PrintWriter(BeforeActions.myObj);
+    out.println(CdfPipelineRunAction.captureRawLogs());
+    out.close();
+  }
+
+  @Then("Validate successMessage is displayed")
+  public void validateSuccessMessageIsDisplayed() {
+    CdfLogActions.validateSucceeded();
+  }
+
+  @Then("Get Count of no of records transferred to BigQuery in {string}")
+  public void getCountOfNoOfRecordsTransferredToBigQueryIn(String tableName) throws IOException, InterruptedException {
+    int countRecords = GcpClient.countBqQuery(CdapUtils.pluginProp(tableName));
+    BeforeActions.scenario.write("**********No of Records Transferred******************:" + countRecords);
+    Assert.assertTrue(countRecords > 0);
```

Comment:
verify actual number of records written, not just > 0

Reply:
Changed assertion.